sigils/reactive.nim

```diff
@@ -0,0 +1,46 @@
+import sigils/signals
+import sigils/slots
+import sigils/core
+
+export signals, slots, core
+
+type
+  Sigil*[T] = ref object of Agent
+    ## Core *reactive* data type for doing reactive style programming
+    ## akin to RXJS, React useState, Svelte, etc.
+    ## 
+    ## This builds on the core signals and slots but provides a
+    ## higher level API for working with propagating values.
+    val*: T
+
+proc changed*[T](r: Sigil[T], val: T) {.signal.}
+  ## core reactive signal type
+
+proc setValue*[T](r: Sigil[T], val: T) {.slot.} =
+  ## default slot action for `changed`
+  r.val = val
+
+proc `<-`*[T](s: Sigil[T], val: T) =
+  if s.val != val:
+    emit s.changed(val)
+
+template reactive*[T](x: T): Sigil[T] =
```

Comment:
Since this is mostly a sigil constructor I'd argue it makes sense to also call this `sigil` rather than `reactive`.
It's shorter and the connection that it constructs a Sigil is more eminent. Could also call it `newSigil` if you want to go fully by nim standards for constructor procs.

Reply:
I'm gonna keep these in `sigils/reactive.nim` for now though. But normally folks would just pull in `import sigils` and get everything.